coroutine-utils/src/main/kotlin/dev/msfjarvis/aps/util/coroutines/RunSuspendCatching.kt

```diff
@@ -0,0 +1,48 @@
+@file:OptIn(ExperimentalContracts::class)
+@file:Suppress("RedundantSuspendModifier")
+
+package dev.msfjarvis.aps.util.coroutines
+
+import com.github.michaelbull.result.Err
+import com.github.michaelbull.result.Ok
+import com.github.michaelbull.result.Result
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+import kotlinx.coroutines.CancellationException
+
+/**
+ * Calls the specified function [block] with [this] value as its receiver and returns its
+ * encapsulated result if invocation was successful, catching any [Throwable] except
+ * [CancellationException] that was thrown from the [block] function execution and encapsulating it
+ * as a failure.
+ */
+public suspend inline fun <V> runSuspendCatching(block: () -> V): Result<V, Throwable> {
```

Comment:
```suggestion
public suspend inline fun <V> runSuspendCatching(block: suspend () -> V): Result<V, Throwable> {
```

Reply:
The block should be suspend I think?

Reply:
IntelliJ threw a warning when I added `suspend` saying that a lambda parameter would use the same continuation.

```
Redundant 'suspend' modifier: lambda parameters of suspend function type uses existing continuation.
```